ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs

```diff
@@ -105,4 +107,15 @@ data TraceForgeEvent blk
   -- 'anachronisticProtocolLedgerView', although we expect this to be
   -- 'TooFarAhead', never 'TooFarBehind'.
   | TraceCouldNotForge SlotNo AnachronyFailure
+
+  -- | We adopted the block we produced
+  | TraceAdoptedBlock SlotNo blk
+
+  -- | We did not adopt the block we produced
```

Comment:
Say that this is very rare, i.e. warrants a warning.

Reply:
We can even be a bit more specific maybe? At least indicate it's not an error either: the block we produced was _valid_ (at least after the changes to this PR), but chain selection decided not to adopt it (so it must have found a better -- longer -- chain, probably due to a concurrent call to `addBlock`?).